### PR TITLE
Argument parser refactoring

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | n
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Bundle/Resources/config/services/expression_language.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <service id="sylius.resource_factory.expression_language" class="Symfony\Component\ExpressionLanguage\ExpressionLanguage" />
+        <service id="sylius.repository.expression_language" class="Symfony\Component\ExpressionLanguage\ExpressionLanguage" />
+        <service id="sylius.routing.expression_language" class="Symfony\Component\ExpressionLanguage\ExpressionLanguage" />
```
Comment (lchrusciel) [inline src/Bundle/Resources/config/services/expression_language.xml:18]: Let's create one service and three aliases

File: src/Bundle/Resources/config/services/expression_language.xml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        <service id="sylius.expression_language.variables_collection.factory" class="Sylius\Component\Resource\Symfony\ExpressionLanguage\VariablesCollection">
+            <argument type="tagged_iterator" tag="sylius.resource_factory_variables" />
+        </service>
+
+        <service id="sylius.expression_language.variables_collection.repository" class="Sylius\Component\Resource\Symfony\ExpressionLanguage\VariablesCollection">
+            <argument type="tagged_iterator" tag="sylius.repository_variables" />
+        </service>
+
+        <service id="sylius.expression_language.variables_collection.routing" class="Sylius\Component\Resource\Symfony\ExpressionLanguage\VariablesCollection">
+            <argument type="tagged_iterator" tag="sylius.routing_variables" />
```
Comment (lchrusciel) [inline src/Bundle/Resources/config/services/expression_language.xml:41]: Where is this tag used?

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @loic425!

File: src/Bundle/Resources/config/services/expression_language.xml (hunk 2 above)
Comment (loic425) [reply src/Bundle/Resources/config/services/expression_language.xml:41]: This tag is never used. This is just the opened door to add custom variables to the argument parser.

File: src/Bundle/Resources/config/services/expression_language.xml (hunk 1 above)
Comment (loic425) [reply src/Bundle/Resources/config/services/expression_language.xml:18]: We'll create our own service to inject custom functions (notFoundOnNull for example)

Comment (loic425) [reply src/Bundle/Resources/config/services/expression_language.xml:18]: Hum, you're right with aliases

**Pull request merged**